### PR TITLE
Deadlock with user

### PR DESCRIPTION
**Pull request by mmoqui**
Take into account the change in the branch deadlock-with-user in Silverpeas-Core.
See [PR 820](https://github.com/Silverpeas/Silverpeas-Core/pull/820)

### CODE REVIEW

File: yellowpages/yellowpages-war/src/main/java/org/silverpeas/components/yellowpages/control/YellowpagesSessionController.java [hunk 1]
```diff
@@ -765,185 +764,123 @@ public String getCurrentSearchCriteria() {
...
+   * @param query
+   * @return list of ContactFatherDetail
+   * @throws FormException
+   * @throws PublicationTemplateException
+   * @
+   */
+  public List<ContactFatherDetail> search(String typeSearch, String query)
+      throws PublicationTemplateException, FormException {
+    List<ContactFatherDetail> result;
+    String _query = query.trim();
```
Comment (silveruser) [inline yellowpages/yellowpages-war/src/main/java/org/silverpeas/components/yellowpages/control/YellowpagesSessionController.java]: MAJOR Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS00117)

**Comment by silveruser** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 2 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WhitePagesSessionController.java#L836](https://github.com/Silverpeas/Silverpeas-Components/blob/29b4eaa258271cb49a342cb9adcfe50db4294afe/whitePages/whitePages-war/src/main/java/org/silverpeas/components/whitepages/control/WhitePagesSessionController.java#L836): Define and throw a dedicated exception instead of using a generic one. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS00112)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

**Comment by silveruser** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WhitePagesSessionController.java#L825](https://github.com/Silverpeas/Silverpeas-Components/blob/541eb27fe5d5bd5cb77d3423df947a32d11524ee/whitePages/whitePages-war/src/main/java/org/silverpeas/components/whitepages/control/WhitePagesSessionController.java#L825): Define and throw a dedicated exception instead of using a generic one. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS00112)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

**Comment by SilverYoCha** [comment]
Merged.
Closed manually because of manual git operations performed during integration.

**Pull request closed without merging**